### PR TITLE
Fixing things up in preparation for larger changes

### LINKED ISSUE #133: Extending the life time of multiple VMs takes too much of time.
Extending the VMs is done one by one, waiting for each job to complete sequentially. Much of time can be saved from broker runtime if done concurrently, for example using asyncio.

### LINKED ISSUE #135: File logs in broker contain terminal colouring
This may be problemattic when displaying as plaintext.

### PR DESCRIPTION
**Pull request by JacobCallahan**
+ Extends are now concurrent by default
+ Color codes are stripped from log files
+ Broker now respects its own debug setting
+ Session command timeouts are now flexible
Fixes #133
Fixes #135

### CODE REVIEW

**Review by mshriver — APPROVED** [review]

File: broker/helpers.py [hunk 1]
```diff
@@ -361,4 +361,26 @@ def handle_keyboardinterrupt(*args):
...
+            unit = "h"
+        if unit == "h":
+            timeout *= 60
+            unit = "m"
+        if unit == "m":
+            timeout *= 60
+            unit = "s"
+        if unit == "s":
+            timeout *= 1000
+    return timeout if isinstance(timeout, int) else 0
```
Comment (tstrych) [inline broker/helpers.py:386]: there was no and still is no new line at the end of the file, this is first time I noticed that github is showing such a thing

**Review by tstrych — APPROVED** [review]
ACK

File: broker/helpers.py (hunk 1 above)
Comment (JacobCallahan) [reply broker/helpers.py:386]: oh weird. good catch!

Comment (JacobCallahan) [reply broker/helpers.py]: and now fixed

**Pull request merged**